js_modules/dagster-ui/packages/ui-core/src/app/AppError.tsx

```diff
@@ -48,28 +47,43 @@ const showNetworkError = async (statusCode: number) => {
   }
 };
 
-export const errorLink = onError((response) => {
-  if (response.graphQLErrors) {
-    const {graphQLErrors, operation} = response;
-    const {operationName} = operation;
-    graphQLErrors.forEach((error) => showGraphQLError(error as DagsterGraphQLError, operationName));
-  }
-  if (response.networkError) {
+export const createErrorLink = (toastOnErrors?: boolean) =>
+  onError((response) => {
+    if (response.graphQLErrors) {
+      const {graphQLErrors, operation} = response;
+      const {operationName} = operation;
+      graphQLErrors.forEach((error) => {
+        if (toastOnErrors) {
+          showGraphQLError(error, operationName);
+        }
+        console.error('[Graphql error]', operationName, error);
+      });
+    }
     // if we have a network error but there is still graphql data
     // the payload should contain a meaningful error for the product to handle
-    const serverError = response.networkError as ServerError;
-    if (serverError.result && typeof serverError.result === 'object') {
+    const serverError = response.networkError;
+
+    if (
+      'response' in response &&
+      response.response &&
+      'data' in response.response &&
+      response.response.data
+    ) {
+      return Observable.from([{data: response.response.data}]);
```

Comment:
i think its worth adding more comment context as to why we have this seemingly goofy stuff to prevent next apollo upgrader from going "wtf" and deleting it